frontend/model/contracts/group.js

```diff
@@ -1871,7 +1808,7 @@ sbp('chelonia/defineContract', {
           // Now, let's see if we had a key request that's been answered
           const sentKeyShares = sbp('chelonia/contract/successfulKeySharesByContractID', state, identityContractID)
           // We received a key share after the last time we left
-          if (sentKeyShares?.[identityContractID]?.[0].height > state.profiles[member].departedHeight) {
+          if (sentKeyShares?.[identityContractID]?.[0].height > state.profiles[memberID].departedHeight) {
```

Comment:
This is part of a whole bunch of new code that was added (although in the diff it doesn't look that way because it's being added by the removal of multiline comments `/* .. */`).

So it's a significant change. Are we sure this code works the way it's supposed to? Doesn't it seem a bit odd for the sideEffect of a `removeMember` action to call `join`? Shouldn't this be done someplace else? And what if there are multiple joins/rejoins, could this code result in duplicate joins being called?

Reply:
This is part of a previous attempt at removing `JOINING_GROUP`. Before things worked correctly without because `JOINING_GROUP` was checking for these cases.

> Doesn't it seem a bit odd for the sideEffect of a removeMember action to call join?

It might seem odd, but this is handling re-joining. In particular, this is addressing the case where the last action is to leave, but we have since requested to re-join (which is a two-step process). I'm not sure if there's another place that's more suitable, as this is part of the contract logic in terms of what should be the next steps.

> And what if there are multiple joins/rejoins, could this code result in duplicate joins being called?

Since this is at the end of the queue, what matters is that the last action we sent was to leave, but we haven't yet sent `inviteAccept`.

Reply:
Why can't we do this near the code that resulted in these key shares being in the contract?

And what specific situation is this (e.g. logging in on a new device... or..?)

Could all of this be elaborated on further in the code comment above?